index.js

```diff
@@ -1,634 +1,465 @@
-				minutes = minutes == '' ? '' : minutes.lpad('0', 2)
-				self.setVariable('minutes', minutes)
-
-				seconds = (jsonData.s < 0 ? '+' : '') + jsonData.s.replace('-', '')
-				seconds = seconds == '' ? '' : seconds.lpad('0', 2)
-				self.setVariable('seconds', seconds)
-
-				self.setVariable('speed', jsonData.speed)
-				self.setVariable('name', jsonData.name)
-				self.setVariable('endTime', jsonData.endTime)
-
-				let tempFg = self.hexToRgb(jsonData.fg.substr(0, 1) + jsonData.fg.substr(3))
-				self.fgColor = self.rgb(tempFg.r, tempFg.g, tempFg.b)
-				let tempBg = self.hexToRgb(jsonData.bg.substr(0, 1) + jsonData.bg.substr(3))
-				self.bgColor = self.rgb(tempBg.r, tempBg.g, tempBg.b)
-
-				self.buttonStates['Fullscreen'] = jsonData['Fullscreen']
-				self.buttonStates['NDI'] = jsonData['NDI']
-				self.buttonStates['Message'] = jsonData['Message']
-				self.buttonStates['STM'] = jsonData['STM']
-				self.buttonStates['Clock'] = jsonData['Clock']
-				self.buttonStates['Pause'] = jsonData['Pause']
-				self.buttonStates['Blackout'] = jsonData['Blackout']
-
-				self.setVariable('nextTimerName', jsonData.nextTimerName)
-				self.setVariable('nextTimerDuration', jsonData.nextTimerDuration)
-
-				self.timers = jsonData.timers
-
-				self.checkFeedbacks('colors', 'status', 'timerName', 'timerDuration', 'timerBackground')
-			} catch (e) {
-				console.error(e.message)
-				return
-			}
-		})
-		self.socket.on('end', function () {
-			console.log('end')
-		})
-	}
-}
+		if (self.socket !== undefined) {
+			self.socket.destroy()
+			delete self.socket
+		}
 
-instance.prototype.config_fields = function () {
-	var self = this
-
-	const dynamicVariableChoices = [];
-	dynamicVariableChoices.push({
-		id: null,
-		label: '(No Variable Selected)'
-	});
-
-	try {
-		this.system.emit('variable_get_definitions', (definitions) =>
-			Object.entries(definitions).forEach(([instanceLabel, variables]) =>
-				variables.forEach((variable) =>
-					dynamicVariableChoices.push({
-						id: `${instanceLabel}:${variable.name}`,
-						label: `${instanceLabel}:${variable.name}`
+		if (self.config.host && self.config.port) {
+			self.socket = new TCPHelper(self.config.host, self.config.port)
+
+			self.socket.on('status_change', (status, message) => {
+				self.log('debug', `Status ${status}, message: ${message}`)
+				self.updateStatus(status)
+			})
+
+			self.socket.on('error', (_err) => {
+				self.updateStatus(InstanceStatus.UnknownError)
+			})
+
+			self.socket.on('connect', () => {
+				self.updateStatus(InstanceStatus.Ok)
+			})
+
+			self.socket.on('data', (data) => {
+				let received = new MessageBuffer('$')
+				received.push(data)
+				let message = received.handleData()
+
+				try {
+					let jsonData = JSON.parse(message)
+					let hours = (jsonData.h < 0 ? '+' : '') + jsonData.h.replace('-', '')
+					hours = hours == '' ? '' : hours.lpad('0', 2)
+					self.setVariableValues({ hours: hours })
```

Comment:
I feel like this `setVariableValues` for hours, as well as the ones for minutes and seconds, could be rolled into the later one that sets speed, name, and endTime

Reply:
I think you are right

Reply:
Done